lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -469,16 +470,41 @@ isSubrangeOf r1 r2 =
                                   Stake Pools
 -------------------------------------------------------------------------------}
 
--- | Represent stake pool identifier. Note that the internal representation is
--- left open currently, until we figure out a better type for those.
-newtype PoolId = PoolId
-    { getPoolId :: Text }
+-- | Represent stake pool identifier.
+newtype PoolId = PoolVRFPubKey { getPoolId :: ByteString }
     deriving (Generic, Eq, Show)
 
 instance NFData PoolId
 
 instance Buildable PoolId where
-    build = build . getPoolId
+    build poolId = mempty
+        <> prefixF 8 poolIdF
+        <> "..."
+        <> suffixF 8 poolIdF
+      where
+        poolIdF = build (toText poolId)
+
+instance ToText PoolId where
+    toText = T.decodeUtf8
+        . convertToBase Base16
+        . getPoolId
+
+-- | VRF PubKey: 32 bytes (ristretto25519) which translates to 64 hex character string
+--   see https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md
+poolIdHexStringLength :: Int
+poolIdHexStringLength = 64
+
+instance FromText PoolId where
+    fromText t
+        | T.length t /= poolIdHexStringLength =
+            Left $ TextDecodingError $
+                "stake pool id invalid: expected " <> show poolIdHexStringLength
+                <> " characters in hex string"
```

Comment:
I think we should print the actual length too for debugability.
```suggestion
                <> " characters in hex string but got "
                <> show (T.length t)
```

```haskell
fromText' @PoolId  "4c43d68b21921034519c36d2475f5adba989bb4465ec12b919faece26ee695233"
Left "stake pool id invalid: expected 64 characters in hex string but got 65"
```

Reply:
good suggestion - added